### PR TITLE
External Network: support allocating static LoadBalancerIP and NodePort

### PR DESCRIPTION
**Pull request by fra98**
# Description

This PR adds the support to allocate static LoadBalancerIP and NodePort.

### CODE REVIEW

File: cmd/liqoctl/cmd/network.go [hunk 1]
```diff
@@ -160,6 +160,10 @@ func newNetworkConnectCommand(ctx context.Context, options *network.Options) *co
 		fmt.Sprintf("Service type of the Gateway Server. Default: %s", gatewayserver.DefaultServiceType))
 	cmd.Flags().Int32Var(&options.ServerPort, "server-port", gatewayserver.DefaultPort,
 		fmt.Sprintf("Port of the Gateway Server. Default: %d", gatewayserver.DefaultPort))
+	cmd.Flags().Int32Var(&options.ServerNodePort, "node-port", 0,
+		"Force NodePort of Gateway Server")
+	cmd.Flags().StringVar(&options.ServerLoadBalancerIP, "load-balancer-ip", "",
+		"Force LoadBalancer IP of Gateway Server")
```
Comment (cheina97) [inline cmd/liqoctl/cmd/network.go:166]: Can you specify that values 0 and "" makes the port and ip random?

File: pkg/liqoctl/network/cluster.go [hunk 2]
```diff
@@ -211,6 +213,25 @@ func (c *Cluster) GetGatewayClient(ctx context.Context, name string) (*networkin
...
+
+		if endpoint.NodePort != nil && *endpoint.NodePort != service.Spec.Ports[0].NodePort {
+			return true
+		}
+	}
+	if endpoint.LoadBalancerIP != nil && *endpoint.LoadBalancerIP != service.Spec.LoadBalancerIP {
+		return true
+	}
+	return false
+}
```
Comment (cheina97) [inline pkg/liqoctl/network/cluster.go:233]: What about using something like this?

https://github.com/kubernetes-sigs/controller-runtime/blob/5f8d96b043ef8ccadc544535415395be336630e8/pkg/controller/controllerutil/controllerutil.go#L216C1-L218C3

File: pkg/liqoctl/rest/gatewayserver/create.go [hunk 3]
```diff
@@ -78,6 +78,8 @@ func (o *Options) Create(ctx context.Context, options *rest.CreateOptions) *cobr
 	cmd.Flags().Var(o.ServiceType, "service-type", fmt.Sprintf("Service type of Gateway Server. Default: %s", DefaultServiceType))
 	cmd.Flags().IntVar(&o.MTU, "mtu", DefaultMTU, "MTU of Gateway Server")
 	cmd.Flags().Int32Var(&o.Port, "port", DefaultPort, "Port of Gateway Server")
+	cmd.Flags().Int32Var(&o.NodePort, "node-port", 0, "Force NodePort of Gateway Server")
+	cmd.Flags().StringVar(&o.LoadBalancerIP, "load-balancer-ip", "", "Force LoadBalancer IP of Gateway Server")
```
Comment (cheina97) [inline pkg/liqoctl/rest/gatewayserver/create.go:82]: Can you specify that values 0 and "" makes the port and ip random?

**Review by cheina97 — APPROVED** [review]

File: pkg/liqoctl/network/cluster.go (hunk 2 above)
Comment (fra98) [reply pkg/liqoctl/network/cluster.go:233]: I don't think we could use it in this case as the two types are different from each other, resulting always in a false equivalency.

**Pull request merged**